xwiki-platform-core/xwiki-platform-annotation/xwiki-platform-annotation-test/xwiki-platform-annotation-test-pageobjects/src/main/java/org/xwiki/annotation/test/po/AnnotatableViewPage.java

```diff
@@ -310,4 +310,17 @@ public int getCommentId(String annotationId)
             "//div[contains(@class, 'annotation')][a[contains(@href,'#" + annotationId + "')]]"))
             .getAttribute("id").replace("xwikicomment_", ""));
     }
+
+    /**
+     * Search for an annotation by its id and return the corresponding web element.
+     *
+     * @param id the id of the annotation to search for
+     * @return the web element of the requested annotation
+     * @since 16.2.0RC1
+     * @since 15.10.7
+     */
+    public WebElement getAnnotationTextById(int id)
+    {
+        return getDriver().findElement(By.cssSelector("span.annotation.ID%d".formatted(id)));
```

Comment:
`#formatted` is only available since Java 15 so not good for a backport to 15.10.x IMO